src/mem/ruby/common/DataBlock.cc

```diff
@@ -73,7 +85,17 @@ DataBlock::clear()
 bool
 DataBlock::equal(const DataBlock& obj) const
 {
-    return !memcmp(m_data, obj.m_data, RubySystem::getBlockSizeBytes());
+    size_t block_bytes = RubySystem::getBlockSizeBytes();
+    // Check that the block contents match
+    if (memcmp(m_data, obj.m_data, block_bytes))
+        return false;
```

Comment:
minor nit: please add parentheses around these if statements.  It will bite us in the butt later if we don't.

Technically not required though.